hibernate-core/src/main/java/org/hibernate/dialect/CockroachDialect.java

```diff
@@ -223,6 +239,23 @@ public void contributeTypes(TypeContributions typeContributions, ServiceRegistry
 				jdbcTypeRegistry.addDescriptorIfAbsent( PostgreSQLJsonJdbcType.INSTANCE );
 			}
 		}
+
+		// Force Blob binding to byte[] for CockroachDB
+		jdbcTypeRegistry.addDescriptor( Types.BLOB, VarbinaryJdbcType.INSTANCE );
+		jdbcTypeRegistry.addDescriptor( Types.CLOB, VarcharJdbcType.INSTANCE );
```

Comment:
```suggestion
		jdbcTypeRegistry.addDescriptor( Types.BLOB, BlobJdbcType.PRIMITIVE_ARRAY_BINDING );
		jdbcTypeRegistry.addDescriptor( Types.CLOB, ClobJdbcType.STRING_BINDING );
```

I think this would be more in-line with what we have elsewhere.

Reply:
I tried that, but it doesn't work on CockroachDB. I changed the descriptor to varbinary because in CockroachDB BLOB is an alias for the BYTES type

Reply:
What's the problem then when using what I proposed?

Reply:
The postgresdriver expects the Blob column to contain a Long (which is the OID for the blob), but CockroachDB stored the Blob object in the column as a byte array. This then leads to this exception: 
```
Caused by: org.postgresql.util.PSQLException: Bad value for type long : \xaced0005737200116a6176612e7574696c2e486173684d61700507dac1c31660d103000246000a6c6f6164466163746f724900097468726573686f6c6478703f4000000000000c7708000000100000000274000464617461740009696d616765646174617400046d6574617400086d6574616461746178
	at app//org.postgresql.jdbc.PgResultSet.toLong(PgResultSet.java:3043)
	at app//org.postgresql.jdbc.PgResultSet.getLong(PgResultSet.java:2216)
	at app//org.postgresql.jdbc.PgResultSet.getBlob(PgResultSet.java:428)
```
(this is from the o.h.o.test.jpa.lob.BlobTest)

Reply:
Yeah, I get that, but when you use the code Suggestion I give it should use the same JDBC methods as for Varchar and Varbinary

Reply:
Apparently, it does not. After applying the suggested code changes, I still get the exact same exception.

Reply:
Btw, your suggested code is what I also initially did. But, as I said, it didn't fix the issue. So that's when I went with the solution of using VarbinaryJdbcType